aries-core/src/js/components/core/Selector/SelectorGroup.js

```diff
@@ -0,0 +1,142 @@
+import React, {
+  useState,
+  useContext,
+  useCallback,
+  Children,
+  useMemo,
+} from 'react';
+import PropTypes from 'prop-types';
+import { Grid, ThemeContext, ResponsiveContext } from 'grommet';
+
+export const SelectorGroupContext = React.createContext({});
+
+const useControlled = ({ prop, defaultProp, onChange = () => {} }) => {
+  const [uncontrolledProp, setUncontrolledProp] = useState(defaultProp);
+  const controlled = prop !== undefined;
+  const value = controlled ? prop : uncontrolledProp;
+  const handleChange = useCallback(onChange, [onChange]);
+
+  const setValue = useCallback(
+    event => {
+      // only update internal value in uncontrolled cases
+      if (!controlled) {
+        setUncontrolledProp(event.value);
+      }
+      handleChange(event);
+    },
+    [controlled, setUncontrolledProp, handleChange],
+  );
+
+  return [value, setValue];
+};
+
+const SelectorGroup = ({
+  children,
+  defaultValue,
+  multiple,
+  onSelect,
+  value,
+  ...rest
+}) => {
+  const size = useContext(ResponsiveContext);
+  const [selectedValue = multiple ? [] : '', setSelectedValue] = useControlled({
+    prop: value,
+    defaultProp: defaultValue,
+    onChange: onSelect,
+  });
+
+  const totalChildren = Children.count(children);
+
+  const handleToggle = useCallback(
+    (event, option) => {
+      const adjustedEvent = event;
+      let nextValue;
+      if (!multiple) {
+        nextValue = option === selectedValue ? '' : option;
+      } else {
+        nextValue = selectedValue.includes(option)
+          ? selectedValue.filter(item => item !== option)
+          : [...selectedValue, option];
+      }
+      adjustedEvent.value = nextValue;
+      setSelectedValue(adjustedEvent);
+    },
+    [multiple, selectedValue, setSelectedValue],
+  );
+
+  const contextValue = useMemo(
+    () => ({
+      multiple,
+      defaultValue,
+      onSelect,
+      value,
+      selectedValue,
+      setSelectedValue,
+      handleToggle,
+    }),
+    [
+      multiple,
+      defaultValue,
+      onSelect,
+      value,
+      selectedValue,
+      setSelectedValue,
+      handleToggle,
+    ],
+  );
+
+  return (
+    <ThemeContext.Extend
+      // theme overrides to remove excess padding
+      value={{
+        checkBox: { pad: 'none' },
+        radioButton: {
+          hover: {
+            background: 'none',
+          },
+          container: {
+            extend: '',
+          },
+          extend: '',
+        },
+      }}
+    >
```

Comment:
Can we remove this theme override since we are using icons now?

Reply:
good call out yes!